pkg/controllermanager/controller/seed/seed_reconcile.go

```diff
@@ -125,35 +122,36 @@ func (r *reconciler) cleanupStaleSecrets(ctx context.Context, gardenClient kuber
 	var fns []flow.TaskFn
 	exclude := sets.NewString(existingSecrets...)
 
-	secrets, err := r.secretLister.Secrets(namespace).List(gardenRoleSelector)
-	if err != nil {
+	secretList := &corev1.SecretList{}
```

Comment:
Can you replace `gardenClient kubernetes.Interface` with `gardenClient client.Client` in this file?